### PR TITLE
Removed cartridge-java from dependencies

### LINKED ISSUE #69: Remove cartridge-java from dependencies
- Update `executeScript` and `executeCommand` methods to execute code viva `execInContainer`.
- Add `executeScriptDecoded` and `executeCommandDecoded` methods to return parsed `yaml` not `String`.
- Remove cartridge-java client from `TarantoolContainerClientHelper` and tests.
- Rewrite tests and add new cases to support new API.
- Remove `io.tarantool.cartridge-driver` dependency.
- Update `org.yaml.snakeyaml` to 2.0 version.

### PR DESCRIPTION
**Pull request by iDneprov**
- Update `executeScript` and `executeCommand` methods to execute code viva `execInContainer`.
- Add `executeScriptDecoded` and `executeCommandDecoded` methods to return parsed `yaml` not `String`.
- Remove cartridge-java client from `TarantoolContainerClientHelper` and tests.
- Rewrite tests and add new cases to support new API.
- Remove `io.tarantool.cartridge-driver` dependency.
- Update `org.yaml.snakeyaml` to 2.0 version.

I haven't forgotten about:
- [x] Tests
- [x] Changelog
- [x] Documentation
- [x] Commit messages comply with the [guideline](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message)
- [x] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:

Closes #69

### CODE REVIEW

**Comment by iDneprov** [comment]
I also need to think about a container builder for SSL | mTLS tests.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Changelog
 
+## [0.6.0] - 2023-05-17
+BREAKING CHANGES
```
Comment (ArtDu) [inline CHANGELOG.md:4]: Not all of them are breaking changes. Look how we did it cartridge-java 
image

File: pom.xml [hunk 2]
```diff
@@ -49,7 +49,7 @@
         <connection>scm:git:git@github.com:tarantool/cartridge-java-testcontainers.git</connection>
         <developerConnection>scm:git:git@github.com:tarantool/cartridge-java-testcontainers.git</developerConnection>
         <url>http://github.com/tarantool/cartridge-java-testcontainers/tree/master</url>
-        <tag>v0.5.4</tag>
+        <tag>v0.6.0</tag>
```
Comment (ArtDu) [inline pom.xml:52]: Don't change it. Maven will change in release phase

File: src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java [hunk 3]
```diff
@@ -1,88 +1,105 @@
...
-        return executeCommand(String.format("dofile('%s')", containerPath));
+        return executeCommand(String.format("dofile('%s')", containerPath), sslIsActive, keyFile, certFile);
     }
 
-    public CompletableFuture<List<?>> executeCommand(String command, Object... arguments) {
+    public <T> T executeScriptDecoded(String scriptResourcePath, Boolean sslIsActive, String keyFile, String certFile) throws IOException, InterruptedException {
+        Container.ExecResult result = executeScript(scriptResourcePath, sslIsActive, keyFile, certFile);
+
+        if (result.getExitCode() != 0) {
+            throw new IllegalStateException(String.format("Executed script %s with exit code %d," +
```
Comment (akudiyar) [inline src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java:50]: I suggest making these command string templates constant variables

**Review by akudiyar — CHANGES_REQUESTED** [review]
LGTM overall, I have few nitpicks though

**Comment by iDneprov** [comment]
Now `TarantoolCartridgeBootstrapFromLuaWithFixedPortsTest`does not start testcontainer

File: src/main/java/org/testcontainers/containers/TarantoolCartridgeContainer.java [hunk 4]
```diff
@@ -1,18 +1,12 @@
...
 import org.testcontainers.images.builder.ImageFromDockerfile;
 
 import java.net.URL;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.concurrent.CompletableFuture;
+import java.util.*;
```
Comment (akudiyar) [inline src/main/java/org/testcontainers/containers/TarantoolCartridgeContainer.java:9]: Please do not use star imports (set up your IDE to not change this for, say, 100 import lines)

File: src/main/java/org/testcontainers/containers/TarantoolContainer.java [hunk 5]
```diff
@@ -74,38 +66,14 @@ public TarantoolContainer(String dockerImageName) {
...
-     * Constructor for {@link TarantoolContainer}
-     *
-     * @param tarantoolImageParams params for cached image creating
-     * @param clientBuilder        client builder with custom client settings for setting up container
-     */
-    public TarantoolContainer(TarantoolImageParams tarantoolImageParams,
-                              TarantoolClientBuilder clientBuilder) {
-        super(TarantoolContainerImageHelper.getImage(tarantoolImageParams));
-        clientHelper = new TarantoolContainerClientHelper(this, clientBuilder);
+        clientHelper = new TarantoolContainerClientHelper(this, "/sdk/tarantoolctl");
```
Comment (akudiyar) [inline src/main/java/org/testcontainers/containers/TarantoolContainer.java:76]: This should be a constant at least, but better that should be either an env variable or the executable should be in the classpath so no explicit path is specified. DO not leave such hardcode, that will not work for all environments

File: src/main/java/org/testcontainers/containers/TarantoolContainer.java [hunk 6]
```diff
@@ -39,6 +37,9 @@ public class TarantoolContainer extends GenericContainer<TarantoolContainer>
     private String password = API_PASSWORD;
     private String host = DEFAULT_HOST;
     private Integer port = DEFAULT_PORT;
+    private Boolean sslIsActive = false;
+    private String keyFile = "";
```
Comment (akudiyar) [inline src/main/java/org/testcontainers/containers/TarantoolContainer.java:41]: I think this should be a default path or name, shouldn't it?

File: src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java [hunk 7]
```diff
@@ -1,88 +1,109 @@
...
+        "Executed command \"%s\" with exit code %d, stderr: \"%s\", stdout: \"%s\"";
 
     TarantoolContainerClientHelper(TarantoolContainerOperations<? extends Container<?>> container) {
         this.container = container;
-        this.clientBuilder = TarantoolClientFactory.createClient()
-                .withRequestTimeout(5000)
-                .withRetryingByNumberOfAttempts(15,
-                        TarantoolRequestRetryPolicies.retryNetworkErrors(),
-                        b -> b.withDelay(100));
+        this.tarantoolctlPath = "tarantoolctl";
```
Comment (akudiyar) [inline src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java:32]: I suggest using an environment variable with default value here

File: src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java [hunk 8]
```diff
@@ -1,88 +1,109 @@
...
         if (!container.isRunning()) {
             throw new IllegalStateException("Cannot execute commands in stopped container");
         }
 
-        return getClient().eval(command, Arrays.asList(arguments));
+        command = command.replace("\"", "\\\"");
+
+        String bashCommand;
+        if (keyFile != "" && certFile != "") {
+            bashCommand = String.format("echo \"print(require('yaml').encode(require('net.box').connect({ uri='%s:%d', " +
```
Comment (akudiyar) [inline src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java:73]: This should be a constant., not an inline string. Also, we may use more proper script formatting for easier reading with backslashes or plus signs.

The same for the subsequent lines

File: src/test/java/org/testcontainers/containers/CartridgeContainerTestUtils.java [hunk 9]
```diff
@@ -15,10 +15,10 @@ private CartridgeContainerTestUtils() {
 
     static public void executeProfileReplaceSmokeTest(TarantoolCartridgeContainer container) throws Exception {
         container.executeCommand(
-                "return profile_replace(...)", Arrays.asList(1, "Ivanov Ivan Ivanovich", 33, 100500)).get();
+                "return profile_replace({1, \"Ivanov Ivan Ivanovich\", 33, 100500})");
 
-        List<?> result = container.executeCommand("return profile_get(...)", 1).get();
+        ArrayList<?> result = container.executeCommandDecoded("return profile_get(1)");
```
Comment (akudiyar) [inline src/test/java/org/testcontainers/containers/CartridgeContainerTestUtils.java:20]: I don't get why we need a more specific type here. Do we use any `ArrayList` methods or properties? Is `ArrayList` always guaranteed as a result (as I see it is a generic parameter so it seems not)?

**Review by akudiyar — CHANGES_REQUESTED** [review]

**Comment by iDneprov** [comment]
SSL and mTLS support will be added in #75

**Comment by iDneprov** [comment]
@ArtDu, @akudiyar, guys, this RP is ready for review.

**Comment by iDneprov** [comment]
@akudiyar, hi! We are planning to merge this PR this week. Will you have time to watch it again?

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,6 +4,19 @@
 - Fix problem if topology isn't applied correctly
 - Bump testcontainers to 1.18.0
 - Move rocks building in build phase
+- Bump testcontainers to 1.18.0
+- Move rocks building in build phase
```
Comment (akudiyar) [inline CHANGELOG.md:8]:
```suggestion
- Move rocks building to the build phase
```

File: src/main/java/org/testcontainers/containers/TarantoolCartridgeContainer.java [hunk 11]
```diff
@@ -119,6 +120,7 @@ public class TarantoolCartridgeContainer extends GenericContainer<TarantoolCartr
     private String directoryResourcePath = SCRIPT_RESOURCE_DIRECTORY;
     private String instanceDir = INSTANCE_DIR;
     private String topologyConfigurationFile;
+    private SslContext sslContext ;
```
Comment (akudiyar) [inline src/main/java/org/testcontainers/containers/TarantoolCartridgeContainer.java:123]:
```suggestion
    private SslContext sslContext;
```

File: src/main/java/org/testcontainers/containers/TarantoolContainer.java [hunk 12]
```diff
@@ -213,6 +167,20 @@ public TarantoolContainer withPassword(String password) {
         return this;
     }
 
+
+    /**
+     * Specify SSL as connection transport. And path to key and cert files inside your container for mTLS connection
+     * Warning! SSL must be set as default transport on your tarantool cluster.
```
Comment (akudiyar) [inline src/main/java/org/testcontainers/containers/TarantoolContainer.java:173]:
```suggestion
     * Warning! SSL must be set as the default transport in your Tarantool cluster.
```

**Review by akudiyar — CHANGES_REQUESTED** [review]
It seems that we are putting too many details to the changelog. These changes can be read from the git history. Leaving internal changes that do not affect customers does not make sense, and it's better for readability to shorten several small changes related to the same feature to one line that describes the meaning of the changes.

File: pom.xml [hunk 13]
```diff
@@ -81,20 +81,21 @@
...
         </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.28</version>
+            <scope>provided</scope>
```
[the remark this answers is not in the document]
Comment (iDneprov) [reply pom.xml:97]: I use it in SslContext class, not in the `test` scope. 
Why you do not recommend using lombok in other scopes?

[the remark this answers is not in the document]
Comment (akudiyar) [reply pom.xml:97]: Using it for a single class with 2 fields and in a library code needs additional justification. We've just removed the driver dependency (which is a good move), why are we adding to the library level a new dependency that is normally used on application level?

**Comment by akudiyar** [comment]
A side note: commits named like "Fix <someones> comments" are not useful for the git history. They need either to be squashed or re-formulated in a way that the message describes what exact changes are made to the code (examples: "rename variable", "pull code literals to constants", "remove/add dependency XXX" etc.)

File: CHANGELOG.md [hunk 14]
```diff
@@ -3,7 +3,14 @@
 ## [Unreleased]
 - Fix problem if topology isn't applied correctly
 - Bump testcontainers to 1.18.0
-- Move rocks building in build phase
+- Move rocks building to the build phase
+- **[breaking change]** Remove io.tarantool.cartridge-driver dependency
+- **[breaking change]** Update executeScript and executeCommand methods to execute code viva execInContainer
```
Comment (dkasimovskiy) [inline CHANGELOG.md:8]:
```suggestion
- **[breaking change]** Update executeScript and executeCommand methods to execute code via execInContainer  
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -3,7 +3,14 @@
 ## [Unreleased]
 - Fix problem if topology isn't applied correctly
 - Bump testcontainers to 1.18.0
-- Move rocks building in build phase
+- Move rocks building to the build phase
+- **[breaking change]** Remove io.tarantool.cartridge-driver dependency
+- **[breaking change]** Update executeScript and executeCommand methods to execute code viva execInContainer  
+  (now it returns yaml string in Container.ExecResult not CompletableFuture)
```
Comment (dkasimovskiy) [inline CHANGELOG.md:9]:
```suggestion
  (now it returns **yaml** string in Container.ExecResult instead of CompletableFuture)
```

**Review by dkasimovskiy — COMMENTED** [review]
Fix changelog, please.

**Review by dkasimovskiy — APPROVED** [review]

**Comment by iDneprov** [comment]
> A side note: commits named like "Fix someone's comments" or "Update filename" are not useful for the git history. 

All commits will be squashed into one before the changes are merged into master. I try not to merge them during reviews to make it easier to keep track of requested changes.

If everything looks good for you, just approve changes, and I'll use "Squash and merge" button.

File: src/test/java/org/testcontainers/containers/Enterprise/TarantoolSslContainerTestEnterprise.java [hunk 16]
```diff
@@ -0,0 +1,62 @@
+package org.testcontainers.containers.Enterprise;
```
Comment (ArtDu) [inline src/test/java/org/testcontainers/containers/Enterprise/TarantoolSslContainerTestEnterprise.java:1]:
```suggestion
package org.testcontainers.containers.enterprise;
```

File: src/main/java/org/testcontainers/containers/TarantoolContainerOperations.java [hunk 17]
```diff
@@ -44,6 +45,13 @@
      */
     String getInstanceDir();
 
+    /**
+     * Get the Tarantool server internal port for connecting the client to
```
Comment (ArtDu) [inline src/main/java/org/testcontainers/containers/TarantoolContainerOperations.java:49]:
```suggestion
     * Get the Tarantool server internal port for client connections
```

File: src/main/java/org/testcontainers/containers/TarantoolCartridgeContainer.java [hunk 18]
```diff
@@ -82,6 +80,9 @@
  * specified in the http_port options, will be exposed.
  *
  * @author Alexey Kuzin
+ * @authorm Artyom Dubinin
```
Comment (ArtDu) [inline src/main/java/org/testcontainers/containers/TarantoolCartridgeContainer.java:83]:
```suggestion
 * @author Artyom Dubinin
```

File: src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java [hunk 19]
```diff
@@ -1,88 +1,141 @@
...
-        String scriptName = Paths.get(scriptResourcePath).getFileName().toString();
-        String containerPath = normalizePath(Paths.get(TMP_DIR, scriptName));
-        container.copyFileToContainer(MountableFile.forClasspathResource(scriptResourcePath), containerPath);
-        return executeCommand(String.format("return dofile('%s')", containerPath));
+        command = command.replace("\"", "\\\"");
+        command = command.replace("\'", "\\\'");
+
+        String bashCommand;
+        // No SSL
+        if (sslContext == null) {
```
Comment (ArtDu) [inline src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java:104]: I think this type of comment code style is more appropriate in this case
```suggestion
        if (sslContext == null) { // No SSL
```

File: src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java [hunk 20]
```diff
@@ -1,88 +1,141 @@
...
+            bashCommand = String.format(COMMAND_TEMPLATE,
+                container.getHost(), container.getInternalPort(),
+                container.getUsername(), container.getPassword(),
+                command
+            );
+        // mTLS
+        } else if (sslContext.getKeyFile() != null && sslContext.getCertFile() != null) {
```
Comment (ArtDu) [inline src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java:111]:
```suggestion
        } else if (sslContext.getKeyFile() != null && sslContext.getCertFile() != null) {  // mTLS
```

File: src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java [hunk 21]
```diff
@@ -1,88 +1,141 @@
...
+            bashCommand = String.format(MTLS_COMMAND_TEMPLATE,
+                container.getHost(), container.getInternalPort(),
+                sslContext.getKeyFile(), sslContext.getCertFile(),
+                container.getUsername(), container.getPassword(),
+                command
+            );
+        // SSL
+        } else {
```
Comment (ArtDu) [inline src/main/java/org/testcontainers/containers/TarantoolContainerClientHelper.java:119]:
```suggestion
        } else {  // SSL
```

**Review by ArtDu — APPROVED** [review]

File: src/test/java/org/testcontainers/containers/enterprise/TarantoolMTlsContainerTestEnterprise.java [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+
+    @BeforeAll
+    public static void setUp() throws Exception {
+        final File dockerfile = new File(
+            TarantoolMTlsContainerTestEnterprise.class.getClassLoader()
+                .getResource("enterprise/Dockerfile").toURI()
+        );
+        final Map<String, String> buildArgs = new HashMap<>();
+        buildArgs.put("DOWNLOAD_SDK_URI", System.getenv("DOWNLOAD_SDK_URI"));
+        buildArgs.put("SDK_VERSION", "tarantool-enterprise-sdk-nogc64-2.10.6-0-r557.linux.x86_64");
```
Comment (dkasimovskiy) [inline src/test/java/org/testcontainers/containers/enterprise/TarantoolMTlsContainerTestEnterprise.java:36]: Я бы вынес версию SDK в константу или settings.

**Review by dkasimovskiy — COMMENTED** [review]
Есть небольшие замечания.
В остальном, можно мержить.

File: src/main/java/org/testcontainers/containers/TarantoolContainer.java [hunk 23]
```diff
@@ -213,6 +167,20 @@ public TarantoolContainer withPassword(String password) {
         return this;
     }
 
+
+    /**
+     * Specify SSL as connection transport. And path to key and cert files inside your container for mTLS connection
```
[the remark this answers is not in the document]
Comment (iDneprov) [reply src/main/java/org/testcontainers/containers/TarantoolContainer.java:172]: I meant, specify transport and path.

**Pull request merged**